addons/point_of_sale/tests/test_pos_basic_config.py

```diff
@@ -928,3 +1007,55 @@ def test_invoice_past_order(self):
+            'orders': [
+                {'pos_order_lines_ui_args': [(self.product_multi_tax, 1)], 'payments': [(self.bank_pm1, 117.7)], 'customer': False, 'is_invoiced': False, 'uid': '00100-010-0001'},
+            ],
+            'journal_entries_before_closing': {},
+            'journal_entries_after_closing': {
+                'session_journal_entry': {
+                    'line_ids': [
+                        {'account_id': self.tax_received_account.id, 'partner_id': False, 'debit': 0, 'credit': 7, 'reconciled': False},
+                        {'account_id': self.tax_received_account.id, 'partner_id': False, 'debit': 0, 'credit': 10.7, 'reconciled': False},
+                        {'account_id': self.sales_account.id, 'partner_id': False, 'debit': 0, 'credit': 100, 'reconciled': False},
+                        {'account_id': self.bank_pm1.receivable_account_id.id, 'partner_id': False, 'debit': 117.7, 'credit': 0, 'reconciled': True},
+                    ],
+                },
+                'cash_statement': [],
+                'bank_payments': [
+                    ((117.7, ), {
+                        'line_ids': [
+                            {'account_id': self.bank_pm1.outstanding_account_id.id, 'partner_id': False, 'debit': 117.7, 'credit': 0, 'reconciled': False},
+                            {'account_id': self.bank_pm1.receivable_account_id.id, 'partner_id': False, 'debit': 0, 'credit': 117.7, 'reconciled': True},
+                        ]
+                    })
+                ],
+            },
+        })
+
+        closed_session = self.pos_session
+        self.assertTrue(closed_session.state == 'closed', 'Session should be closed.')
+
+        order_to_invoice = closed_session.order_ids[0]
+        test_customer = self.env['res.partner'].create({'name': 'Test Customer'})
+
+        with freeze_time(fields.Datetime.now() + relativedelta(days=2)):
+            # Create a new session after 2 days
+            self.open_new_session(0)
+            # Invoice the uninvoiced order
+            order_to_invoice.write({'partner_id': test_customer.id})
+            order_to_invoice.action_pos_order_invoice()
+            # Check the invoice for the lines
+            self.assertTrue(order_to_invoice.account_move, 'Invoice should be created.')
+            self.assertRecordValues(order_to_invoice.account_move.line_ids, [
+                {'account_id': self.sales_account.id, 'partner_id': test_customer.id, 'debit': 0, 'credit': 100, 'reconciled': False},
+                {'account_id': self.tax_received_account.id, 'partner_id': test_customer.id, 'debit': 0, 'credit': 7, 'reconciled': False},
+                {'account_id': self.tax_received_account.id, 'partner_id': test_customer.id, 'debit': 0, 'credit': 10.7, 'reconciled': False},
+                {'account_id': self.receivable_account.id, 'partner_id': test_customer.id, 'debit': 117.7, 'credit': 0, 'reconciled': True},
```

Comment:
-shorter to check the balance instead of debit/credit
- checking the partner is useless imo

Reply:
Okay, makes sense, updated